### PR TITLE
Add eslint (take two)

### PR DESCRIPTION
**Pull request by nathsimpson**
Install ESLINT, and a basic GitHub Action, as well as resolve any eslint errors.

Basically reimplementing #19 on the right branch

### CODE REVIEW

File: .github/workflows/nodejs.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Node.js CI
```
Comment (elisechant) [inline .github/workflows/nodejs.yaml:1]: shall we rename the workflow from `nodejs.yaml` to say `merge-checks.yaml`?

File: .github/workflows/nodejs.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x]
```
Comment (elisechant) [inline .github/workflows/nodejs.yaml:11]: shall we simply set one node version? perhaps you also want to add this version to the package.json `engines`.

File: website/components/article.js [hunk 3]
```diff
@@ -1,8 +1,10 @@
 /** @jsx jsx */
 import { jsx, css } from "@emotion/core";
 import { Fragment } from "react";
-import { Index, items } from "../pages";
+import { Index } from "../pages";
 import Item from "../components/item";
+
+// eslint-disable-next-line no-unused-vars
```
Comment (elisechant) [inline website/components/article.js:7]: can this be refactored out by removing the unused props?

File: .github/workflows/nodejs.yaml (hunk 2 above)
Comment (nathsimpson) [reply .github/workflows/nodejs.yaml:11]: Which version should we use? 15?

File: package.json [hunk 4]
```diff
@@ -9,10 +9,17 @@
     "docs": "yarn workspace @thinkmill/monorepo-docs",
     "website:build": "yarn website build",
     "website:start": "yarn docs:format && yarn website start",
-    "docs:format": "yarn docs format"
+    "docs:format": "yarn docs format",
+    "lint": "yarn eslint . --ext .ts,.tsx,.js"
```
Comment (elisechant) [inline package.json:13]: try removing the `.` before the extensions, `"lint": "yarn eslint . --ext ts,tsx,js"`

File: .eslintrc [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    "ecmaVersion": 2018,
+    "sourceType": "module"
+  },
+  "plugins": ["react", "prettier"],
+  "rules": {
+    "prettier/prettier": "error",
+    "react/jsx-uses-react": "error",
+    "react/jsx-uses-vars": "error"
+  },
+  "overrides": [
```
Comment (elisechant) [inline .eslintrc:21]: Since it's not being used, lets delete `overrides?

File: .github/workflows/nodejs.yaml (hunk 2 above)
Comment (elisechant) [reply .github/workflows/nodejs.yaml:11]: Yep, general rule newest stable one, 12 is good.

**Comment by elisechant** [comment]
Hey @nathsimpson this is a pretty valuable contribution 👍  good job. Just a couple of tweaks otherwise LGTM.

File: website/components/article.js [hunk 6]
```diff
@@ -1,8 +1,10 @@
 /** @jsx jsx */
 import { jsx, css } from "@emotion/core";
 import { Fragment } from "react";
-import { Index, items } from "../pages";
+import { Index } from "../pages";
```
Comment (elisechant) [inline website/components/article.js:4]: This might be unrelated to your change, but thoughts being it is odd that a component is importing a page.

**Review by elisechant — APPROVED** [review]
:shipit:

**Pull request merged**